node/services/service.go

```diff
@@ -78,11 +77,9 @@ func HeaderStore(ds datastore.Batching) (header.Store, error) {
 // BlockService constructs new block.Service.
 func BlockService(
 	lc fx.Lifecycle,
-	fetcher *core.BlockFetcher,
 	store ipld.DAGService,
-	broadcaster header.Broadcaster,
 ) *block.Service {
-	service := block.NewBlockService(fetcher, store, broadcaster)
+	service := block.NewBlockService(store)
```

Comment:
You're removing the fetcher and broadcaster here, but aren't they needed for fetching and broadcasting blocks, not headers?

Reply:
No, block reconstruction will be done by fetching shares via `ShareService` instead and reconstructed / stored within `BlockService`.